validator/sawtooth_validator/journal/consensus/poet/poet_consensus/poet_block_publisher.py

```diff
@@ -0,0 +1,279 @@
+        signup_info = \
+            SignupInfo.create_signup_info(
+                poet_enclave_module=self._poet_enclave_module,
+                validator_address=block_header.signer_pubkey,
+                originator_public_key_hash=public_key_hash,
+                most_recent_wait_certificate_id=wait_certificate_id)
+
+        # Create the validator registry payload
+        payload = \
+            vr_pb.ValidatorRegistryPayload(
+                verb='register',
+                name='validator-{}'.format(block_header.signer_pubkey[-8:]),
+                id=block_header.signer_pubkey,
+                signup_info=vr_pb.SignUpInfo(
+                    poet_public_key=signup_info.poet_public_key,
+                    proof_data=signup_info.proof_data,
+                    anti_sybil_id=signup_info.anti_sybil_id),
+                block_num=block_header.block_num)
+        serialized = payload.SerializeToString()
+
+        # Create the address that will be used to look up this validator
+        # registry transaction.  Seems like a potential for refactoring..
+        validator_entry_address = \
+            PoetBlockPublisher._validator_registry_namespace + \
+            hashlib.sha256(block_header.signer_pubkey.encode()).hexdigest()
+
+        # Create a transaction header and transaction for the validator
+        # registry update amd then hand it off to the batch publisher to
+        # send out.
+        addresses = \
+            [validator_entry_address,
+             PoetBlockPublisher._validator_map_address]
+
+        header = \
+            txn_pb.TransactionHeader(
+                signer_pubkey=block_header.signer_pubkey,
+                family_name='sawtooth_validator_registry',
+                family_version='1.0',
+                inputs=addresses,
+                outputs=addresses,
+                dependencies=[],
+                payload_encoding="application/protobuf",
+                payload_sha512=hashlib.sha512(serialized).hexdigest(),
+                batcher_pubkey=block_header.signer_pubkey,
+                nonce=time.time().hex().encode()).SerializeToString()
+        signature = \
+            signing.sign(header, self._batch_publisher.identity_signing_key)
+
+        transaction = \
+            txn_pb.Transaction(
+                header=header,
+                payload=serialized,
+                header_signature=signature)
+
+        self._batch_publisher.send([transaction])
+
+        LOGGER.info('Register validator: Name=%s', payload.name)
+        LOGGER.info('Register validator: ID=%s', payload.id)
+        LOGGER.info(
+            'Register validator: PoET Public Key=%s',
+            payload.signup_info.poet_public_key)
+
+        # HACER: Once we have the consensus state implemented, we can
+        # store this information in there.  For now, we will store in the
+        # class so it persists for the lifetime of the validator.
+        PoetBlockPublisher._sealed_signup_data = \
```

Comment:
Like 'FXM', I would like to leave in place so in next phase it is easy/obvious where I need to attack.